### PR TITLE
Tailscale package

### PR DESCRIPTION
**Pull request by nshalman**
I think this is probably most of what is needed, but I could use some help with testing and fixing any bugs/quirks of the package.

If I start figuring this stuff out, I hope to use it as a stepping stone to packaging wireguard-go and wireguard-tools to further my goal of getting all of this work upstreamed into the appropriate projects.

### CODE REVIEW

File: build/tailscale/files/network-tailscale.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+<service_bundle type="manifest" name="export">
+  <service name="ooce/network/tailscale" type="service" version="0">
+    <create_default_instance enabled="true"/>
```
Comment (nshalman) [inline build/tailscale/files/network-tailscale.xml:5]: This should probably be false (suggested by @jclulow on IRC and I agree.)

File: build/tailscale/local.mog [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+
+# Copyright 2021 OmniOS Community Edition (OmniOSce) Association.
```
Comment (nshalman) [inline build/tailscale/local.mog:11]: Wrong year

File: build/tailscale/files/network-tailscale.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <dependency name="network" grouping="require_all" restart_on="error" type="service">
+      <service_fmri value="svc:/milestone/network:default"/>
+    </dependency>
+    <dependency name="filesystem" grouping="require_all" restart_on="error" type="service">
+      <service_fmri value="svc:/system/filesystem/local"/>
+    </dependency>
+    <method_context>
+      <method_credential group="root" user="root"/>
+    </method_context>
+    <exec_method name="start" type="method" exec="/opt/ooce/sbin/tailscaled" timeout_seconds="60"/>
```
Comment (nshalman) [inline build/tailscale/files/network-tailscale.xml:16]: I'd prefer not to hardcode this path if possible... Are there good examples of how to template this kind of thing?

Comment (citrus-it) [reply build/tailscale/files/network-tailscale.xml:16]: Thanks for the pull request, we're all getting back to things after the holidays so sorry for the delay in looking at this. Somebody will be along to do a proper review in a day or two.

You can use variables in SMF manifests in conjunction with a call to `xform`; `openldap` is one example.
The `xform` function uses the same transformation variables as the `.mog` files so you can do something like this in the build.sh

```bash
XFORM_ARGS+="
    -DPREFIX=${PREFIX#/}
    -DPROG=$PROG
"

...

xform files/$PROG.xml > $TMPDIR/$PROG.xml
install_smf ooce $PROG.xml
```

Comment (nshalman) [reply build/tailscale/files/network-tailscale.xml]: Thank you!!

**Comment by oetiker** [comment]
Hi @nshalman, thanks for your contribution! @citrus-it and @hadfl will probably provide you with some feedback in the next few days.

File: build/tailscale/build.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        tailscaled
+    popd >/dev/null
+}
+
+install() {
+    pushd $TMPDIR/$BUILDDIR > /dev/null
+    mkdir -p $DESTDIR/$PREFIX/sbin
+    mkdir -p $DESTDIR/$PREFIX/bin
+    mv tailscaled $DESTDIR/$PREFIX/sbin/tailscaled
+    ln -s ../sbin/tailscaled $DESTDIR/$PREFIX/bin/tailscale
```
Comment (hadfl) [inline build/tailscale/build.sh:52]: I'd probably move this to `local.mog`

```
link path=$(PREFIX)/bin/tailscale target=../sbin/tailscaled
```

File: build/tailscale/files/tailscale.xml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
```
Comment (hadfl) [inline build/tailscale/files/tailscale.xml:1]: please run `tools/tidy-manifests` on the mainfest to format it consistently with other manifests

File: build/tailscale/build.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    -DSERVICE=$PKG
+"
+set_arch 64
+set_gover 119
+
+build() {
+    pushd $TMPDIR/$BUILDDIR > /dev/null
+    export CGO_ENABLED=0
+    export GOOS=illumos
+    bash -x ./build_dist.sh --box ./cmd/tailscaled
```
Comment (hadfl) [inline build/tailscale/build.sh:39]: maybe prefix this (and other) commands with `logcmd` and add `|| logerr "<error message>"` in case it exits non-zero

**Review by citrus-it — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**